include/mega/node.h

```diff
@@ -33,6 +33,8 @@ struct MEGA_API NodeCore
     NodeCore();
     ~NodeCore();
 
+    MEGA_DISABLE_COPY_MOVE(NodeCore)
```

Comment:
Since the aim is to make NewNode moveable, we should not disable move in its base class here, but allow it to be moved also, and have NewNode's move constructor use NodeCore's move constructor.    You can get all that for free (along with implicitly disabling copy), and remove the need for the destructor here by protecting the only non-POD member attrstring by a unique_ptr instead of a raw pointer.  (rule of 0)